src/webgpu/shader/execution/expression/binary/bitwise_shift.spec.ts

```diff
@@ -0,0 +1,311 @@
+export const description = `
+Execution Tests for the bitwise shift binary expression operations
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { GPUTest } from '../../../../gpu_test.js';
+import { i32, scalarType, TypeU32, u32 } from '../../../../util/conversion.js';
+import { allInputSources, CaseList, run } from '../expression.js';
+
+import { binary } from './binary.js';
+
+export const g = makeTestGroup(GPUTest);
+
+function is_unsiged(type: string) {
+  return type === 'u32';
+}
+
+const bitwidth = 32;
+
+// Returns true if e1 << e2 is valid for const evaluation
```

Comment:
Is there a convenient single location in the spec that defines the rules being enforced here that you can reference, or is this more a distillation of experience about how multiple rules interact?

Reply:
These are the rules spelled out under `https://www.w3.org/TR/WGSL/#bit-expr` for the shift left and right rows in the table (I wish I could link more directly to them). I put this URL as the `specUrl` for each test below. If you feel I should repeat the URL here, I can do that as well.

Reply:
That should be sufficient